src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/OpportunitiesOrchestrator.cs

```diff
@@ -120,12 +126,61 @@ public async Task<OpportunitySummaryViewModel> GetOpportunitySummaryViewModel(Op
             };
         }
 
+        public OpportunitySummaryViewModel GetOpportunitySummaryViewModel(
+            IEnumerable<string> sectors,
+            IEnumerable<string> jobRoles,
+            IEnumerable<string> levels,
+            IEnumerable<ReferenceDataItem> allSectors,
+            IEnumerable<ReferenceDataItem> allJobRoles,
+            IEnumerable<ReferenceDataItem> allLevels,
+            int amount,
+            bool isNamePublic,
+            string dasAccountName,
+            string encodedPledgeId)
+        {
+            string sectorList = sectors.ToReferenceDataDescriptionList(allSectors);
+            string jobRoleList = jobRoles.ToReferenceDataDescriptionList(allJobRoles);
+            string levelList = levels.ToReferenceDataDescriptionList(allLevels, descriptionSource: x => x.ShortDescription);
+
+            DateTime dateTime = _dateTimeService.UtcNow;
+
+            return new OpportunitySummaryViewModel()
+            {
+                Amount = amount,
+                Description = isNamePublic ? $"{dasAccountName} ({encodedPledgeId})" : "A levy-paying business",
+                JobRoleList = jobRoleList,
+                LevelList = levelList,
+                SectorList = sectorList,
+                YearDescription = dateTime.ToTaxYearDescription(),
+            };
+        }
+
         public async Task<ApplyViewModel> GetApplyViewModel(ApplicationRequest request)
         {
+            var application = await RetrieveCacheItem(request.CacheKey);
             var opportunityDto = await _opportunitiesService.GetOpportunity(request.PledgeId);
 
+            var emailAddresses = new List<string>();
```

Comment:
I don't follow the logic here. Surely when the cache item gets set it's just a list of strings, with no empty entries? In which case, we wouldn't need this?

Reply:
This has been changed now.